src/rust/src/properties_reader.rs

```diff
@@ -27,6 +27,61 @@ impl FgbFeature {
     pub fn geometry(&self) -> Option<Geometry> {
         self.fbs_feature().geometry()
     }
+
+    fn dimension(&self) -> geo_traits::Dimensions {
+        match (self.header().has_z(), self.header().has_m()) {
+            (true, true) => geo_traits::Dimensions::Xyzm,
+            (true, false) => geo_traits::Dimensions::Xyz,
+            (false, true) => geo_traits::Dimensions::Xym,
+            (false, false) => geo_traits::Dimensions::Xy,
+        }
+    }
+
+    /// Access the underlying geometry, returning an object that implements
+    /// [geo_traits::GeometryTrait].
+    ///
+    /// Note that this method only supports the core geometry types supported by [geo_traits], and
+    /// will panic on curve geometries.
+    pub fn geometry_trait_impl(&self) -> Result<Option<impl geo_traits::GeometryTrait + use<'_>>> {
```

Comment:
I learned in https://github.com/kylebarron/wkb that this `use<'_>` syntax requires Rust 1.82. https://rust-lang.github.io/rfcs/3617-precise-capturing.html

This signature would be nice to use as it lets all of `geo_trait_impl` be a private module. I don't see a current MSRV:
https://github.com/kylebarron/flatgeobuf/blob/7e0f27e50ddde197164afcb532132796c775609a/src/rust/Cargo.toml#L2-L13

Thoughts?

Reply:
Adding an MSRV 1.82 is fine for me.